app/src/main/java/io/homeassistant/companion/android/sensors/ProximitySensorManager.kt

```diff
@@ -0,0 +1,87 @@
+package io.homeassistant.companion.android.sensors
+
+import android.content.Context
+import android.content.Context.SENSOR_SERVICE
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener
+import android.hardware.SensorManager.SENSOR_DELAY_NORMAL
+import io.homeassistant.companion.android.domain.integration.SensorRegistration
+import kotlin.math.roundToInt
+
+class ProximitySensorManager : SensorManager, SensorEventListener {
+    companion object {
+
+        private const val TAG = "ProximitySensor"
+        private val proximitySensor = SensorManager.BasicSensor(
+            "proximity_sensor",
+            "sensor",
+            "Proximity Sensor"
+        )
+        private var proximityReading: String = "unavailable"
+        lateinit var mySensorManager: android.hardware.SensorManager
+    }
+
+    override val name: String
+        get() = "Proximity Sensors"
+
+    override val availableSensors: List<SensorManager.BasicSensor>
+        get() = listOf(proximitySensor)
+
+    override fun requiredPermissions(): Array<String> {
+        return emptyArray()
+    }
+
+    override fun getSensorData(
+        context: Context,
+        sensorId: String
+    ): SensorRegistration<Any> {
+        return when (sensorId) {
+            proximitySensor.id -> getProximitySensor(context)
+            else -> throw IllegalArgumentException("Unknown sensorId: $sensorId")
+        }
+    }
+
+    private fun getProximitySensor(context: Context): SensorRegistration<Any> {
+
+        mySensorManager = context.getSystemService(SENSOR_SERVICE) as android.hardware.SensorManager
+
+        val proximitySensors = mySensorManager.getDefaultSensor(Sensor.TYPE_PROXIMITY)
+        if (proximitySensors != null) {
+            mySensorManager.registerListener(
+                this,
+                proximitySensors,
+                SENSOR_DELAY_NORMAL)
+
+            // Some devices only report 2 values, one of which is the max range so lets account for those devices
+            if (proximitySensors.maximumRange.roundToInt() == 5) {

```

Comment:
We should just check if the value for max == sensor value.  I didn't see anywhere in the documentation that says five is the magic value.  We should just check if they are equal to determine if they're far.

Reply:
Ok so I can change this to `proximitySensors.maximumRange.roundToInt().toString() == proximityReading` so we verify the maximum range is the value we just got, this also simplifies the condition statement :)

I got the number 5 from : https://developer.android.com/guide/topics/sensors/sensors_position

At the bottom of the section we see them call out `5` in the note

```
Note: Some proximity sensors return binary values that represent "near" or "far." In this case, the sensor usually reports its maximum range value in the far state and a lesser value in the near state. Typically, the far value is a value > 5 cm, but this can vary from sensor to sensor. You can determine a sensor's maximum range by using the getMaximumRange() method.
```

Reply:
@JBassett so the problem with checking if the max range == sensor value is that all devices will only report the binary values then.  I think since google mentioned "typically the far value is a value > 5 cm" is a good indication of what to expect here.  Otherwise we can just keep things binary which doesn't make much sense if the device supports a real reading.  I am ok either way here since my device only reports binary values.  Do you have any other suggestions for this?